tests/phpunit/Civi/Funding/EventSubscriber/FundingCase/FundingCasePermissionsCacheClearSubscriberTest.php

```diff
@@ -94,7 +100,116 @@ public function testPreContact(): void {
+    static::assertNull(FundingCasePermissionsCache::get(FALSE)->execute()->single()['permissions']);
+  }
+
+  public function testOnPreGroupContact(): void {
+    $fundingProgram = FundingProgramFixture::addFixture();
+    $fundingCaseType = FundingCaseTypeFixture::addFixture();
+    $contact1 = ContactFixture::addIndividual();
+    $contact2 = ContactFixture::addIndividual();
+
+    $group1 = GroupFixture::addFixture();
+    $group2 = GroupFixture::addFixture();
+
+    $fundingCase1 = FundingCaseFixture::addFixture(
+      $fundingProgram->getId(),
+      $fundingCaseType->getId(),
+      $contact1['id'],
+      $contact1['id']
+    );
+    $fundingCase2 = FundingCaseFixture::addFixture(
+      $fundingProgram->getId(),
+      $fundingCaseType->getId(),
+      $contact1['id'],
+      $contact1['id']
+    );
+    FundingCaseContactRelationFixture::addFixture($fundingCase1->getId(), Relationship::getEntityName(), [
+      'relationshipTypeIds' => [],
+      'contactTypeIds' => [],
+      'groupIds' => [$group1['id']],
+    ], ['permission']);
+    FundingCaseContactRelationFixture::addFixture($fundingCase2->getId(), Relationship::getEntityName(), [
+      'relationshipTypeIds' => [],
+      'contactTypeIds' => [],
+      'groupIds' => [$group2['id']],
+    ], ['permission']);
+
+    FundingCasePermissionsCacheFixture::add($fundingCase1->getId(), $contact1['id'], FALSE, ['test']);
+
+    // Adding contact to group should affect cache.
+    $groupContact = GroupContactFixture::addFixtureWithGroupId($group1['id'], $contact1['id']);
+    static::assertNull(FundingCasePermissionsCache::get(FALSE)->execute()->single()['permissions']);
+
+    // Clear permissions and add new one.
+    FundingCasePermissionsCache::delete(FALSE)->addWhere('id', 'IS NOT NULL')->execute();
+    FundingCasePermissionsCacheFixture::add($fundingCase1->getId(), $contact1['id'], FALSE, ['test']);
+
+    // Removing contact from group should affect cache.
+    GroupContact::delete(FALSE)->addWhere('id', '=', $groupContact['id'])->execute();
+    static::assertNull(FundingCasePermissionsCache::get(FALSE)->execute()->single()['permissions']);
+
+    // Clear permissions and add new one.
+    FundingCasePermissionsCache::delete(FALSE)->addWhere('id', 'IS NOT NULL')->execute();
+    FundingCasePermissionsCacheFixture::add($fundingCase1->getId(), $contact1['id'], FALSE, ['test']);
+    FundingCasePermissionsCacheFixture::add($fundingCase1->getId(), $contact2['id'], FALSE, ['test']);
+
+    $groupContact = GroupContactFixture::addFixtureWithGroupId($group1['id'], $contact1['id']);
+
+    // Updating GroupContact should affect cache.
+    $groupContact['contact_id'] = $contact2['id'];
+    $groupContact['group_id'] = $group2['id'];
+    GroupContact::update(FALSE)->setValues($groupContact)->execute();
+
+    static::assertSame([NULL, NULL], FundingCasePermissionsCache::get(FALSE)->execute()->column('permissions'));
+  }
```

Comment:
perhaps also test scenarios where permission cache is not supposed to be affected?